connectors/src/connectors/confluence/index.ts

```diff
@@ -107,8 +110,70 @@ export async function updateConfluenceConnector(
     connectionId?: NangoConnectionId | null;
   }
 ): Promise<Result<string, ConnectorsAPIErrorResponse>> {
-  console.log({ connectorId, connectionId });
-  throw new Error("Not implemented");
+  const connector = await Connector.findOne({
+    where: {
+      id: connectorId,
+    },
+  });
+  if (!connector) {
+    logger.error({ connectorId }, "Connector not found.");
+    return new Err({
+      error: {
+        message: "Connector not found",
+        type: "connector_not_found",
+      },
+    });
+  }
+
+  if (connectionId) {
+    const { connectionId: oldConnectionId } = connector;
+
+    const currentCloudInformation = await ConfluenceConfiguration.findOne({
+      attributes: ["cloudId"],
+      where: {
+        connectorId,
+      },
+    });
+
+    const newConnection = await nango_client().getConnection(
+      getRequiredNangoConfluenceConnectorId(),
+      connectionId,
+      false
+    );
+
+    const confluenceAccessToken = newConnection?.credentials?.access_token;
+    const newConfluenceCloudInformation = await getConfluenceCloudInformation(
+      confluenceAccessToken
+    );
+
+    // Change connection only if "cloudId" matches.
+    if (
+      newConfluenceCloudInformation &&
+      currentCloudInformation &&
+      newConfluenceCloudInformation.id === currentCloudInformation.cloudId
+    ) {
+      await connector.update({ connectionId });
+
+      nangoDeleteConnection(
+        oldConnectionId,
+        getRequiredNangoConfluenceConnectorId()
+      ).catch((e) => {
+        logger.error(
+          { error: e, oldConnectionId },
+          "Error deleting old Nango connection"
+        );
+      });
+    } else {
+      return new Err({
```

Comment:
We can delete the new Nango connection that we're not using?

Reply:
Good catch!